### ISSUE TITLE
Support for dmabuf between corundum and a GPU (Direct p2p between GPU and FPGA)

### ISSUE THREAD

**Issue by dandaly** [corundum/corundum#227]
Hi,
  I was looking through the codebase to see what the approach would be to enable direct data transfer between a GPU and an FPGA running Corundum. All off-the-shelf GPUs (Nvidia, AMD, Intel) support dmabuf for transferring data and while original GPUDirect relied on RDMA, Google has shown how to get DMA working without RDMA: 
  - https://netdevconf.info/0x17/sessions/talk/device-memory-tcp.html
  - https://github.com/google/tcpgpudmarxd
  - https://github.com/google/nccl-plugin-gpudirecttcpx).
 tl;dr they map transfers to sockets (using NCCL) and sockets to queues (using the RX device manager) to do data transfers. They use TCP running on the CPU to make the transfer reliable, and it runs completely outside of the NIC (Corundum in this case).

Seems like it would be straight-forward to support dmabuf on Corundum, has this come up as an interesting feature add already or is it something brand new?
